src/System.IO/tests/TextWriter/TextWriterTests.netcoreapp.cs

```diff
@@ -52,5 +54,68 @@ public void WriteLineCharSpanTest()
                 Assert.Equal(new string(rs.Span) + tw.NewLine, tw.Text);
             }
         }
+
+        [Theory]
+        [MemberData(nameof(GetStringBuilderTestData))]
+        public void WriteStringBuilderTest(bool isSynchronized, string testData)
+        {
+            using (CharArrayTextWriter ctw = NewTextWriter)
+            {
+                TextWriter tw = isSynchronized ? TextWriter.Synchronized(ctw) : ctw;
+                tw.Write(new StringBuilder(testData));
+                tw.Flush();
+                Assert.Equal(testData, ctw.Text);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(GetStringBuilderTestData))]
+        public void WriteLineStringBuilderTest(bool isSynchronized, string testData)
+        {
+            using (CharArrayTextWriter ctw = NewTextWriter)
+            {
+                TextWriter tw = isSynchronized ? TextWriter.Synchronized(ctw) : ctw;
+                tw.WriteLine(new StringBuilder(new string(testData)));
+                tw.Flush();
+                Assert.Equal(testData + tw.NewLine, ctw.Text);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(GetStringBuilderTestData))]
+        public async void WriteAsyncStringBuilderTest(bool isSynchronized, string testData)
+        {
+            using (CharArrayTextWriter ctw = NewTextWriter)
+            {
+                TextWriter tw = isSynchronized ? TextWriter.Synchronized(ctw) : ctw;
+                await tw.WriteAsync(new StringBuilder(testData));
+                tw.Flush();
+                Assert.Equal(testData, ctw.Text);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(GetStringBuilderTestData))]
+        public async void WriteLineAsyncStringBuilderTest(bool isSynchronized, string testData)
+        {
+            using (CharArrayTextWriter ctw = NewTextWriter)
+            {
+                TextWriter tw = isSynchronized ? TextWriter.Synchronized(ctw) : ctw;
+                await tw.WriteLineAsync(new StringBuilder(testData));
+                tw.Flush();
+                Assert.Equal(testData + tw.NewLine, ctw.Text);
+            }
+        }
+
+        // Generate data for TextWriter.Write* methods that take a stringBuilder.  
+        // We test both the synchronized and unsynchronized variation, on strinbuilder swith 0, small and large values.    
+        public static IEnumerable<object[]> GetStringBuilderTestData()
+        {
+            foreach (string testData in new string[] { "", new string(TestDataProvider.CharData), new string(TestDataProvider.LargeData) })
+            {
+                foreach (bool isSynchronized in new bool[] { true, false })
+                    yield return new object[] { isSynchronized, testData };
```

Comment:
Nit: it'd be nice to be consistent with braces, either both foreach having braces or neither foreach having braces.